telemetry/config.go

```diff
@@ -41,6 +41,12 @@ type Config struct {
 	MetricsURLOverride string
 	// SpansURLOverride overrides the spans endpoint if not not empty.
 	SpansURLOverride string
+	// Product is added to the User-Agent header to track usage. eg.
+	// "NewRelic-Go-OpenCensus"
+	Product string
+	// ProductVersion is added to the User-Agent header to track usage. eg.
+	// "0.1.0".
+	ProductVersion string
```

Comment:
Just throwing this out there, what if we combined these two fields? Include in the doc comment what it should look like?

Additionally, can we remove the "why" part of the two doc comments here? The why could change and not sure people would want to add these fields if they think they're only used by NR to track them.

Reply:
I prefer it with two fields because that separates data from formatting:  As a result, the consumer doesn't need to worry about "product/version" string correctness.

I have removed the "why" and force pushed.  Great idea!

Reply:
With regard to including guidelines on what the product and product version should look like, I'm planning to open a PR in the [exporter specs](https://github.com/newrelic/newrelic-exporter-specs) to help define just that.

My initial thoughts is that they need to satisfy [`product`](https://tools.ietf.org/html/rfc7231#section-5.5.3) definition, but we probably want to have more clarity regarding reflecting the code base or application. Planning to discuss in slack.